### PR TITLE
Medical record transaction and acl rules

### PR DESCRIPTION
**Pull request by l3lyu**

### CODE REVIEW

File: app/lib/logic.js [hunk 1]
```diff
@@ -18,26 +18,72 @@
...
+    const assetRegistry = await getAssetRegistry(tx.record.getFullyQualifiedType());
+    await assetRegistry.update(tx.record)
 }
+
+/**
+ * Check if the participant could read medical record
+ * mr: sepcified medical record
+ * p: current participant
+ */
+function check(mr, p) {
```
Comment (Liang-Hsuan) [inline app/lib/logic.js]: Maybe change it to other meaningful name like `checkParticipantAuthorizedToViewMedicalRecord()`

File: app/lib/logic.js [hunk 2]
```diff
@@ -18,26 +18,72 @@
...
+    await assetRegistry.update(tx.record);
+}
+
+/**
+ * Authorize a Particitant to allow read access to a Medical Record
+ * @param {org.example.mynetwork.AuthorizeMedicalRecord} authorizeMedicalRecord
+ * @transaction
+ */
+async function authorizeMedicalRecord(tx) {
+    const authorizedParties = [...tx.record.authorizedParties];
```
Comment (zydc2010) [inline app/lib/logic.js]: [...tx.record.authorizedParties, tx.authorizedParty]

File: app/permissions.acl [hunk 3]
```diff
@@ -13,53 +13,100 @@
...
- * Sample access control list.
+ * Access control for medical records
  */
-rule EverybodyCanReadEverything {
-    description: "Allow all participants read access to all resources"
-    participant: "org.example.mynetwork.SampleParticipant"
-    operation: READ
-    resource: "org.example.mynetwork.*"
-    action: ALLOW
+rule ServiceProviderCreateCreateMedRec {
```
Comment (Liang-Hsuan) [inline app/permissions.acl]: `CreateCreate` 🤔

File: app/lib/logic.js (hunk 2 above)
Comment (zydc2010) [reply app/lib/logic.js]: no push needed

File: app/lib/logic.js (hunk 1 above)
Comment (Tan-He) [reply app/lib/logic.js]: You got it

File: app/lib/logic.js (hunk 2 above)
Comment (Tan-He) [reply app/lib/logic.js]: +1

File: app/lib/logic.js [hunk 4]
```diff
@@ -18,26 +18,72 @@
...
+async function createMedicalRecord(tx) {
+    const factory = getFactory();
+    const namespace = 'org.example.mynetwork';
+
+    const record = factory.newResource(namespace, 'MedicalRecord', tx.medicalRecordId);
+    record.issueDate = tx.issueDate;
+    record.detail = tx.detail;
+    record.patient = tx.patient;
+    record.authorProvider = tx.authorProvider
+    record.authorizedParties = []
```
[the remark this answers is not in the document]
Comment (Tan-He) [reply app/lib/logic.js]: ParseException: Expected "-->", "@", "o", "optional", "}", comment, end of line or whitespace but "d" found.Composer(ParseException)

I don't think you could set default like that for an array

File: app/permissions.acl (hunk 3 above)
Comment (Tan-He) [reply app/permissions.acl]: Create CreateMedRec Transaction🤔 cant think of a better name

**Pull request merged**